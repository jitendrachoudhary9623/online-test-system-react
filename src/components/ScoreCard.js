import React, { Component } from 'react';

export default class ScoreCard extends Component {

    constructor(props) {
        super(props);

    }

    render() {
        return (
            <div>
                <div className="row">
                    <div className="col s12">
                        <div className="card">
                            <div className="card-content center-align">
                                <span className="card-title">Your Score for this test is</span>
                                <h3>{this.props.score}</h3>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

