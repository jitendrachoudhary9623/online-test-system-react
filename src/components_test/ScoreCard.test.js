import React, { Component } from 'react';
import ScoreCard from "../components/client/ScoreCard.component"
import { shallow } from 'enzyme'

describe('UI Test for ScoreCard', () => {
    var score = "10";
    let snapshot = shallow(
        <ScoreCard score={score} />
    );
    it('Score Card to be defined', () => {
        expect(ScoreCard).toBeDefined();
    });

    it('Score Card should render correctly', () => {
        expect(snapshot).toMatchSnapshot();
    });


    it('Score card score should be displayed properly', () => {
        expect(snapshot.find('h3').text()).toEqual(score);
    });

});