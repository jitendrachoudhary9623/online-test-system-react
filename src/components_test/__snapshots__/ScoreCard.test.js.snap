exports[`UI Test for ScoreCard Score Card should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScoreCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row">
        <div
                className="col s12">
                <div
                        className="card">
                        <div
                                className="card-content center-align">
                                <span
                                        className="card-title">
                                        Your Score for this test is
                                </span>
                                <h3 />
                        </div>
                </div>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col s12">
          <div
                    className="card">
                    <div
                              className="card-content center-align">
                              <span
                                        className="card-title">
                                        Your Score for this test is
                              </span>
                              <h3 />
                    </div>
          </div>
</div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card">
            <div
                        className="card-content center-align">
                        <span
                                    className="card-title">
                                    Your Score for this test is
                        </span>
                        <h3 />
            </div>
</div>,
          "className": "col s12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card-content center-align">
              <span
                            className="card-title">
                            Your Score for this test is
              </span>
              <h3 />
</div>,
            "className": "card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card-title">
                  Your Score for this test is
</span>,
                <h3 />,
              ],
              "className": "card-content center-align",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Your Score for this test is",
                  "className": "card-title",
                },
                "ref": null,
                "rendered": "Your Score for this test is",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row">
          <div
                    className="col s12">
                    <div
                              className="card">
                              <div
                                        className="card-content center-align">
                                        <span
                                                  className="card-title">
                                                  Your Score for this test is
                                        </span>
                                        <h3 />
                              </div>
                    </div>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col s12">
            <div
                        className="card">
                        <div
                                    className="card-content center-align">
                                    <span
                                                className="card-title">
                                                Your Score for this test is
                                    </span>
                                    <h3 />
                        </div>
            </div>
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card">
              <div
                            className="card-content center-align">
                            <span
                                          className="card-title">
                                          Your Score for this test is
                            </span>
                            <h3 />
              </div>
</div>,
            "className": "col s12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="card-content center-align">
                <span
                                className="card-title">
                                Your Score for this test is
                </span>
                <h3 />
</div>,
              "className": "card",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="card-title">
                    Your Score for this test is
</span>,
                  <h3 />,
                ],
                "className": "card-content center-align",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Your Score for this test is",
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": "Your Score for this test is",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
